movieApp/MovieListViewController.swift

```diff
@@ -0,0 +1,85 @@
+import UIKit
+import PureLayout
+import MovieAppData
+
+class MovieListViewController: UIViewController, MovieNavigationDelegate, UICollectionViewDelegate  {
+    
+    private var collectionView: UICollectionView!
+    private let movieUseCase: MovieUseCaseProtocol = MovieUseCase()
+    
+   
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupMovieList()
+       
+    }
+    
+    
+    func navigateToMovieDetails(with movieID: Int) {
+           let movieDetailsViewController = MovieDetailsViewController(movieID: movieID)
+           navigationController?.pushViewController(movieDetailsViewController, animated: true)
+    }
```

Comment:
This is a perfect example of a function which should be placed inside your `AppRouter`. Idea of an AppRouter is to handle all routing inside your app.